example/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/longhorn/upgrade-responder/client"
+)
+
+const (
+	// example.com should be replaced by your Upgrade Responder server address
+	upgradeResponderServerAddress = "https://example.com/v1/checkupgrade"
+	queryPeriod                   = time.Hour
+
+	version           = "v0.8.1"
+	kubernetesVersion = "v1.19.1"
+)
+
+func main() {
+	done := make(chan struct{})
+
+	go periodicallyCheckUpgrade(done)
+
+	registerShutdownChannel(done)
+	<-done
+}
+
+func periodicallyCheckUpgrade(done chan struct{}) {
+	ticker := time.NewTicker(queryPeriod)
+	defer ticker.Stop()
+	upgradeResponderClient := client.NewUpgradeResponderClient(upgradeResponderServerAddress)
```

Comment:
Can we wrap the periodic check directly into the library?

E.g. the client only needs to call:

```
checker.SetInterval(1 * time.Hour) [Optional]
checker.Start(<url>, upgradeRequester)
checker.Stop()
```

Define `upgradeRequester` as an interface, with the following functions:
```
func GetCurrentVersion() string {}
func GetExtraInfo() map[string]string {}
func ProcessUpgradeResponse(UpgradeResponse) 
```

So it can be hooked into the client directly without the user need to manually deal with periodic checks.

Reply:
I think we should leave user to handle the periodic check because:
* The extraInfo may change between the upgrade check request calls. For example, if users upgrade Kubernetes patch version, the `kuberntesVersion` is changed. If we let user handle he periodic check, they can set the extraInfo before each request and solve this scenario. 
* The client logic is simpler and user can understand it quickly

Reply:
GetExtraInfo() is a call defined by the caller. They can change as demands.

The logic is simple but what's the benefit for user to handle it vs client package handle it?

Reply:
Thank you. I wrap the periodic check into the library